pkg/server/tls/authority/authority.go

```diff
@@ -383,6 +400,13 @@ func (d *DynamicAuthority) regenerateCA(ctx context.Context, s *corev1.Secret) e
 				cmmeta.TLSCAKey:         certBytes,
 			},
 		}, metav1.CreateOptions{})
+		if err != nil && apierrors.IsAlreadyExists(err) {
+			// another controller has created the secret, we expect a watch event
+			// to trigger another call to ensureCA
+			d.log.V(logf.DebugLevel).Info("Failed to create new root CA Secret, another controller has created it, waiting for watch event")
```

Comment:
I think this operation should be leader election aware to avoid the race condition, but this is an improvement of the current code!

Reply:
There might be a bit of a race, but there will always be convergence with the current implementation.
Maybe we can make transitions more efficient (and eg. reduce downtime) by making it leader election aware.